logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java

```diff
@@ -0,0 +1,143 @@
+package alluxio.logserver;
+
+import alluxio.AlluxioRemoteLogFilter;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.log4j.Hierarchy;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PropertyConfigurator;
+import org.apache.log4j.spi.LoggerRepository;
+import org.apache.log4j.spi.LoggingEvent;
+import org.apache.log4j.spi.RootLogger;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+
+/**
+ * Reads {@link org.apache.log4j.spi.LoggingEvent} objects from remote logging
+ * clients and writes the objects to designated log files. For each logging client,
+ * logging server creates a {@link AlluxioLog4jSocketNode} object and starts
+ * a thread serving the logging requests of the client.
+ */
+public class AlluxioLog4jSocketNode implements Runnable {
+  private static final org.slf4j.Logger LOG =
+      org.slf4j.LoggerFactory.getLogger(AlluxioLog4jSocketNode.class);
+  private final AlluxioLogServerProcess mLogServerProcess;
+  private final Socket mSocket;
+
+  /**
+   * Constructor of {@link AlluxioLog4jSocketNode}.
+   *
+   * Callers construct AlluxioLog4jSocketNode instances, passing the ownership of the socket
+   * parameter. From now on, the AlluxioLog4jSocketNode is responsible for closing the socket.
+   *
+   * @param process main log server process
+   * @param socket client socket from which to read {@link org.apache.log4j.spi.LoggingEvent}
+   */
+  public AlluxioLog4jSocketNode(AlluxioLogServerProcess process, Socket socket) {
+    mLogServerProcess = Preconditions.checkNotNull(process,
+        "The log server process cannot be null.");
+    mSocket = Preconditions.checkNotNull(socket, "Client socket cannot be null");
+  }
+
+  @Override
+  public void run() {
+    LoggerRepository hierarchy = null;
+    LoggingEvent event;
+    Logger remoteLogger;
+    try (ObjectInputStream objectInputStream = new ObjectInputStream(
+        new BufferedInputStream(mSocket.getInputStream()))) {
+      while (!Thread.currentThread().isInterrupted()) {
+        try {
+          event = (LoggingEvent) objectInputStream.readObject();
+        } catch (ClassNotFoundException e1) {
+          throw new RuntimeException("Class of serialized object cannot be found.", e1);
+        } catch (IOException e1) {
+          throw new RuntimeException("Cannot read object from stream due to I/O error.", e1);
+        }
+        if (hierarchy == null) {
+          hierarchy = configureHierarchy(
+              event.getMDC(AlluxioRemoteLogFilter.REMOTE_LOG_MDC_APPENDER_NAME_KEY).toString());
+        }
+        remoteLogger = hierarchy.getLogger(event.getLoggerName());
+        if (event.getLevel().isGreaterOrEqual(remoteLogger.getEffectiveLevel())) {
```

Comment:
what is the benefit of checking this?

Reply:
We had a discussion offline, and decided to get rid of this checking by making the log server log whatever `LoggingEvent`s received from log clients.